src/button.js

```diff
@@ -2,11 +2,16 @@ import React, { Component } from 'react';
 
 class Button extends Component {
   render() {
+    const { id, state, toggleButton } = this.props;
+
     return (
-      <div className="Button">
+      <div
+        onClick={() => toggleButton(id)}
```

Comment:
Finally we call that function that operates on the state allll the way back up there in the App. As soon as the state changes, everything that depends on it down the tree re-renders. That means all we have to worry about at the button level is changing the state. The button will render as if for the first time, just with new props.